Dockerfile.rocm.ubi

```diff
@@ -3,133 +3,107 @@ ARG BASE_UBI_IMAGE_TAG=9.4
+ARG PYTHON_VERSION
+
+ENV VIRTUAL_ENV=/opt/vllm
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
 
-ENV BUILD_TARGET='rocm'
+RUN --mount=type=cache,target=/root/.cache/pip \
+ microdnf install -y --setopt=install_weak_deps=0 --nodocs \
+    python${PYTHON_VERSION}-devel \
+    python${PYTHON_VERSION}-pip \
+    python${PYTHON_VERSION}-wheel && \
+    python${PYTHON_VERSION} -m venv $VIRTUAL_ENV && \
+    pip install -U pip wheel uv
 
-USER root
 
+FROM base AS rocm_base
 ENV ROCM_VERSION=6.1.2
 
-# Set up ROCm repository and install necessary packages
-
-RUN echo "[amdgpu]" > /etc/yum.repos.d/amdgpu.repo && \
-echo "name=amdgpu" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "baseurl=https://repo.radeon.com/amdgpu/${ROCM_VERSION}/rhel/9.4/main/x86_64/" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "enabled=1" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "priority=50" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "gpgcheck=1" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "gpgkey=https://repo.radeon.com/rocm/rocm.gpg.key" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "[ROCm-${ROCM_VERSION}]" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "name=ROCm${ROCM_VERSION}" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "baseurl=https://repo.radeon.com/rocm/rhel9/${ROCM_VERSION}/main" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "enabled=1" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "priority=50" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "gpgcheck=1" >> /etc/yum.repos.d/amdgpu.repo && \
-echo "gpgkey=https://repo.radeon.com/rocm/rocm.gpg.key" >> /etc/yum.repos.d/amdgpu.repo
+RUN printf "[amdgpu]\n\
+name=amdgpu\n\
+baseurl=https://repo.radeon.com/amdgpu/${ROCM_VERSION}/rhel/9.4/main/x86_64/\n\
+enabled=1\n\
+priority=50\n\
+gpgcheck=1\n\
+gpgkey=https://repo.radeon.com/rocm/rocm.gpg.key\n\
+[ROCm-${ROCM_VERSION}]\n\
+name=ROCm${ROCM_VERSION}\n\
+baseurl=https://repo.radeon.com/rocm/rhel9/${ROCM_VERSION}/main\n\
+enabled=1\n\
+priority=50\n\
+gpgcheck=1\n\
+gpgkey=https://repo.radeon.com/rocm/rocm.gpg.key" > /etc/yum.repos.d/amdgpu.repo
+
 
 RUN microdnf -y update && \
-    microdnf -y install rocm hipcc git which && \
+    microdnf -y install \
+        rocm-hip-libraries rocm-hip-runtime \
+        miopen-hip && \
     microdnf clean all
 
-WORKDIR /workspace
 
-##################################################################################################
+FROM rocm_base as rocm_devel
+ARG PYTORCH_ROCM_ARCH
 
-FROM rocm-base as python-install
-ARG PYTHON_VERSION
+ENV CCACHE_DIR=/root/.cache/ccache
 
-ENV VIRTUAL_ENV=/opt/vllm
-ENV PATH="$VIRTUAL_ENV/bin:$PATH"
-RUN microdnf install -y --setopt=install_weak_deps=0 --nodocs \
-    python${PYTHON_VERSION}-devel python${PYTHON_VERSION}-pip python${PYTHON_VERSION}-wheel && \
-    python${PYTHON_VERSION} -m venv $VIRTUAL_ENV --system-site-packages && \
-    $VIRTUAL_ENV/bin/pip install --no-cache -U pip wheel && \
+RUN rpm -ivh https://dl.fedoraproject.org/pub/epel/epel-release-latest-9.noarch.rpm && \
+    rpm -ql epel-release && \
```

Comment:
do we need epel and -ql listing?

Reply:
We need epel for ccache, the `-ql` listing is not required